_drafts/wsl2-as-a-firmware-development-environment.md

```diff
@@ -0,0 +1,485 @@
+
+This is a summary intended to guide you on the trade-offs that come with
+the use of various development environments.  It is by no means exhaustive and I'd
+welcome further feedback in the comments!  Some Zephyr benchmarks were devised
+in order to quantify performance of some common Zephyr toolchains.
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## Features and Limitations
+
+|                                  | Ubuntu 24.04 (Kernel 6.8) | Windows 11 | VMWare Workstation Pro 17 | WSL2 2.2.4.0 (Kernel 5.15) |
+| -------------------------------- | ------------------------- | ---------- | ------------------------- | -------------------------- |
+| Graphics                         | ✅                         | ✅          | ✅<sup>1</sup>             | ✅                          |
+| Network                          | ✅                         | ✅          | ✅                         | ✅                          |
+| USB                              | ✅                         | ✅          | ✅                         | ✅                          |
+| BLE                              | ✅                         | ✅          | ✅                         | ✅<sup>2</sup>              |
+| Performance<sup>3</sup>          | 100%                      | ~50%       | ~40% -> ~88%<sup>4</sup>  | ~88%                       |
+| Developer Experience<sup>5</sup> | ⭐⭐⭐⭐                      | ⭐⭐         | ⭐⭐⭐                       | ⭐⭐⭐⭐                       |
+
+> <sup>1</sup> VMWare Workstation Pro, when run headless, has the typical "X forwarding" setup
+> curve.  You can choose to emulate the graphics instead of forwarding
+> if you are OK with the overhead and latency.
+> 
+> <sup>2</sup> Not supported by default, discussed [below](#ble).
+> 
+> <sup>3</sup> % of bare metal Ubuntu 24.04 performance, based on a Zephyr benchmark, discussed [below](#performance).
+> 
+> <sup>4</sup> VMWare Workstation Pro 17 tested very poorly, but I suspect that 
+> it is because I was unable to get it to use AMD-V.  Of course, AMD-V not working
+> while WSL2 is enabled is another negative for VMWare here!  Nevertheless, I
+> suspect that it would be similar to WSL2 performance if AMD-V / Intel VT-x are working,
+> so I gave it "up to ~88%".
+> 
+> <sup>5</sup> The author's personal opinion, discussed [below](#developer-experience).
+
+### Graphics
+
+If you've previously setup X forwarding in WSL2, you'll be happy to learn that this is no longer
+necessary.  Linux GUIs "Just Work" in WSL2!
+
+### Network
+
+It’s no surprise that WSL2 can do anything you need over the network, since one of its
+primary use cases is web development.
+
+### USB
+
+[usbipd-win](https://github.com/dorssel/usbipd-win) is
+[officially recommended by Microsoft](https://learn.microsoft.com/en-us/windows/wsl/connect-usb)
+and provides a stable foundation for USB forwarding including automatic attach
+by both port and device.
```

Comment:
This is a tricky one, because I think that the USBIPD documentation is slightly out of date there.  I've used USBIPD for at least two years.  If someone has Windows 11, they're guaranteed to have a new enough WSL2, and Windows 10 is end of life soon.

I think that I need to specify use of Windows 11 🤔

Reply:
Actually the page you linked includes a section for [prerequisites](https://learn.microsoft.com/en-us/windows/wsl/connect-usb#prerequisites) so I think we're covered!